drivers/adc/ad7606/ad7606.c

```diff
@@ -1197,6 +1199,10 @@ static int32_t ad7606_axi_init(struct ad7606_dev *device,
 	if (ret != 0)
 		goto error;
 
+	ret = no_os_pwm_init(&axi->trigger_pwm_desc, axi_init->trigger_pwm_init);
```

Comment:
i think this makes the PWM usage mandatory while at the same time there is a convst gpio in the init_param / device driver desc 

it's very good that you added the pwm method but there should be a proper way for the user to use either method without being forced to use a particular one

Reply:
it makes sense then

Reply:
> also, i think (but i'd need to check with the HDL guys) in the HDL design, the CONVST pin is wired to the CONVST pin to drive the conversion;

typo when writing here; apologies;
CONVST pin is wired to the PWM/HDL pin to drive the conversion;